File: Innopolis/digidivisible.py
import math
b, n = map(int, input().split())
t = int(input())
while t > 0:
	t -= 1
	pos = input()
	cur = []
	for x in range(b):
		if pos[x] == '1':
			cur.append(x)
	for i in range(1, n):
		p = math.pow(b, i)
		temp = []
		for j in range(b):
			if pos[j] == '0':
				continue
			for k in cur:
				if k%j != 0:
					continue
				val = True
				for x in str(k):
					if (j*p)%int(x) != 0:
						val = False
				if not val:
					continue
				temp.append(int(j*p+k))
		cur = []
		for j in temp:
			print(j)
			cur.append(j)
	print(len(cur))

File: Innopolis/digidivisible.cpp
#include <iostream>
#include <cmath>
#include <vector>
#include <string>
using namespace std;
int main () {
	int b, n;
	cin >> b >> n;
	vector<int> opts;
	for (int i=pow(b, n-1); i<pow(b, n); i++) {
		int temp = i;
		bool pos = true;
		while (temp > 0) {
			int dig = temp%b;
			temp /= b;
			if (dig == 0 || i%dig != 0) {
				pos = false;
				break;
			}
		}
		if (pos) opts.push_back(i);
	}
	int t;
	cin >> t;
	while (t--) {
		char val[b];
		for (int i=0; i<b; i++) {
			cin >> val[i];
		}
		int o = 0;
		for (auto i: opts) {
			int temp = i;
			bool pos = true;
			while (temp > 0) {
				int dig = temp%b;
				temp /= b;
				if (val[dig] == '0') {
					pos = false;
					break;
				}
			}
			if (pos) o++;
		}
		cout << o << endl;
	}
}